src/probnum/diffeq/odefiltsmooth/diffusions.py

```diff
@@ -0,0 +1,120 @@
+"""Diffusion calibrations.
+
+Maybe move this to `statespace` or `filtsmooth` in the future, but for
+now, it is well placed in `diffeq`.
+"""
+
+
+import abc
+from typing import Union
+
+import numpy as np
+
+from probnum import random_variables
+from probnum.type import FloatArgType
+
+DiffusionType = Union[FloatArgType]
+"""Acceptable diffusion types are -- in principle -- floats and arrays of shape (d,).
+In other words, everything that behaves well with `cov1 + diffusion*cov2` and
+`chol1 + sqrt(diffusion) * chol2`.
+For now, let's only use floats, because it is not clear to me how to best implement the square-root behaviour.
+"""
+
+
+class Diffusion(abc.ABC):
+    r"""Interface for diffusion models :math:`\sigma: \mathbb{R} \rightarrow \mathbb{R}^d` and their calibration."""
+
+    def __repr__(self):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def __call__(self, t) -> DiffusionType:
+        """Evaluate the diffusion."""
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def __getitem__(self, idx):
+        """Get the i-th diffusion from the list."""
+        pass
```

Comment:
these are not used yet (though implemented for constant diffusions already, but I think they will be useful when trying to resolve issues such as #186

Reply:
I'm a bit conflicted about having both `__call__` and `__getitem__`, since the first suggests very general functions as diffusions while the second only makes sense with our choice of step functions. What about only having `__call__`?

Reply:
I think in principle only `__call__` would do most of the work. Though I am note sure whether this is compatible with all the `find_index` stuff in `KalmanPosterior`. However, we can leave it at just `__call__` for now and add a `__getitem__` whenever we see that it is necessary?

Reply:
That sounds good :+1: As you say, if we see that we do need a `__getitem__` we can still add it.